v2/pkg/catalog/loader/loader.go

```diff
@@ -376,3 +379,49 @@ func workflowContainsProtocol(workflow []*workflows.WorkflowTemplate) bool {
 	}
 	return false
 }
+
+func (store *Store) GetUncoverTargetsFromMetadata(delay, limit int) chan string {
+	templatePaths := store.config.Catalog.GetTemplatesPath(store.finalTemplates)
+	ret := make(chan string)
+	go func() {
+		var wg sync.WaitGroup
+		for _, templatePath := range templatePaths {
+			template, err := templates.Parse(templatePath, store.preprocessor, store.config.ExecutorOptions)
+			if err != nil {
+				continue
+			}
+			for k, v := range template.Info.Metadata {
+				var engine []string
+				var query []string
+				switch k {
+				case "shodan-query":
+					engine = append(engine, "shodan")
+				case "fofa-query":
+					engine = append(engine, "fofa")
+				case "censys-query":
+					engine = append(engine, "censys")
+				case "quake-query":
+					engine = append(engine, "quake")
+				case "hunter-query":
+					engine = append(engine, "hunter")
+				case "zoomeye-query":
+					engine = append(engine, "zoomeye")
+				default:
+					continue
+				}
+				query = append(query, fmt.Sprintf("%v", v))
+				wg.Add(1)
```

Comment:
we should add some sort of control on max engines that'll be ran concurrently. probbably using sizedwaitgroup as we do for other stuff

Reply:
used sizedwaitgroup with a max 50 as a limit. Do you have any opinion on the limit?

Reply:
i think 10 would be better or